src/main/java/org/openpnp/machine/reference/ReferenceCamera.java

```diff
@@ -205,12 +228,26 @@ protected BufferedImage transformImage(BufferedImage image) {
             else {
                 flipCode = flipX ? 0 : 1;
             }
-            Mat dst = new Mat();
-            Core.flip(mat, dst, flipCode);
-            mat = dst;
+            Core.flip(mat, mat, flipCode);
         }
         
-        image = OpenCvUtils.toBufferedImage(mat);
+        
+        { double scale = zoom; if (scale<0) { scale=-scale; }
+	  if(scale>1.0) {
+		int w=(int)((double)mat.width()/zoom+.5);
+		int h=(int)((double)mat.height()/zoom*ratio+.5);
+		int x=(mat.width()-w)/2;
+                int y=(mat.height()-h)/2;
+          	Mat img = new Mat(mat,new Rect(x,y,w,h));
+          	if(zoom<0.) {
+    			Imgproc.resize(img, img, mat.size());
+          	}
```

Comment:
@cri-s I am testing this now and the zoom doesn't seem to be doing anything. Looking at this code, I think that you are missing an else clause at this line. Did you want to resize mat at some point?